### PR TITLE
integration test for searchflight

### PR DESCRIPTION
**Pull request by Alvtag**

### CODE REVIEW

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java [hunk 1]
```diff
@@ -18,17 +14,14 @@
...
 
-import java.io.OutputStream;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.*;
 
-import static com.guestlogix.travelercorekit.utilities.UrlHelper.urlEncodeUTF8;
-
 
 public class Router {
```
Comment (Alvtag) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: I changed Router.java to use a private getSessionToken that handles the edge case of a session's Token being null.
I don't think this will happen in the app, since the lib is initialized early on; but in the instrumented test the API call happens very quickly, before the token is fetched.

.build() with a null token value seems to be fine; the task naturally fetches a new token.

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java [hunk 2]
```diff
@@ -382,4 +373,12 @@ private static Error transformError(Error error) {
             return error;
         }
     }
+
+    private static String getSessionToken(Session session) {
+        Token token = session.getToken();
+        if (null == token) {
+            Log.w(TAG, "Session token is null");
```
Comment (Alvtag) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: I'm using android.util.Log to be consistent with the rest of the class; but should we use TravelerLog.java throughout this class?

File: traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        Context instrumentationContext = InstrumentationRegistry.getInstrumentation().getContext();
+        Traveler.initialize(TRAVELER_API_KEY, instrumentationContext);
+    }
+
+    @Test
+    public void flightSearchTest() throws Exception {
+        Date date =  Calendar.getInstance().getTime();
+        FlightQuery query = new FlightQuery("AC100", date);
+        CountDownLatch latch = new CountDownLatch(1);
+        final List<Error> errorList = new ArrayList<>();
```
Comment (ata-n) [inline traveler_core_kit/src/androidTest/java/com/guestlogix/travelercorekit/TravelerTest.java]: Lets create an `Expectation` class for this, instead of tracking success and failure by the count of errors accumulated in an array

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java (hunk 2 above)
Comment (ata-n) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: Nah, TravelerLog is deprecated

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java (hunk 1 above)
Comment (ata-n) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: Lets talk about this. There shouldnt be a need for this. null should go fine and then refetch the token.

Comment (Alvtag) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: The problem is that when `session.getToken()` resolves to null, `.getValue()` causes NPE

Comment (Alvtag) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: will change to passing in TOKEN rather than its VALUE as discussed
